### DISCUSSION TITLE
stumpy for irregular time series

### DISCUSSION THREAD

**Discussion by evoisec** [TDAmeritrade/stumpy#570] [General]
From the documentation of the matrix profile and stumpy itself, i gather that stumpy is directly applicable to irregularly spaced (the time periods between the data points can vary, rather than be constant) time series - is that correct?

**Comment by seanlaw** [comment]
@evoisec Thanks for the question and welcome to the STUMPY community. So, matrix profiles actually don't use "time" for anything. It just takes in a list of ordered floating point numbers and computes the matrix profile for it. It is left up to the user to determine if that ordering is "meaningful". So, if it is irregularly spaced, STUMPY doesn't complain and will still produce a matrix profile but, as they say, "garbage in, garbage out". Sometimes, it's okay for, say, an IoT device to record something roughly every 30 seconds but with some error in that it might be only roughly every 30 seconds and STUMPY doesn't know any of that . It just takes the values that were recorded without regard to the actual time. It is really a sequence.

I hope that helps.

**Comment by evoisec** [comment]
Thanks Sean, great software, with great documentation and also video presentations and as this forum also shows - proactive and prompt  support.

I wish great success to your project - it is something really unique

**Reply by Sceeto** [reply]
So very interesting...thanks, Sean.
